crates/volta-core/src/monitor.rs

```diff
@@ -42,11 +42,21 @@ impl Monitor {
 
 fn spawn_process(command: &str) -> Option<Child> {
     command.split(' ').take(1).next().and_then(|executable| {
-        let child = create_command(executable)
-            .args(command.split(' ').skip(1))
-            .stdin(Stdio::piped()) // JSON data is sent over stdin
-            // .stdout(Stdio::piped()) // let the plugin write to stdout for now
-            .spawn();
+        // Only show stdout and stderr of spawned process in dev mode
+        let child = if cfg!(debug_assertions) {
+            create_command(executable)
+                .args(command.split(' ').skip(1))
+                .stdin(Stdio::piped()) // JSON data is sent over stdin
+                .spawn()
+        } else {
+            create_command(executable)
+                .args(command.split(' ').skip(1))
+                .stdin(Stdio::piped()) // JSON data is sent over stdin
+                .stdout(Stdio::null()) // Pipe stdout to nowhere
+                .stderr(Stdio::null()) // Pipe stdout to nowhere
+                .spawn()
+        };
```

Comment:
Since the only difference between the two cases is the `stdout` and `stderr` calls, I think we can DRY this slightly to make it more clear by creating the command and applying the changes individually instead of chaining them. Something like:

```rust
let mut command = create_command(executable);
command.args(...);
command.stdin(...);

#[cfg(debug_assertions)]
{
    command.stdout(...);
    command.stderr(...);
}

match command.spawn() {
```

I haven't fully worked out if something like that works with the return values, but the key thing is that you don't _have_ to chain all of the `Command` methods, you can call them individually.

Reply:
Cool got it thanks!